explorer/deploy/webAppCreator.ts

```diff
@@ -543,4 +568,74 @@ class WebsiteStep extends WebAppCreatorStepBase {
         }
     }
 
+    //Implements new Service principal model for ACR container registries while maintaining old admin enabled use
+    private async loginCredentials(): Promise<void> {
+        if (this._serverPassword && this._serverUserName) { return; }
+
+        const client = new ContainerRegistryManagementClient(this.azureAccount.getCredentialByTenantId(this._imageSubscription.tenantId), this._imageSubscription.subscriptionId);
+        const resourceGroup: string = this._registry.id.slice(this._registry.id.search('resourceGroups/') + 'resourceGroups/'.length, this._registry.id.search('/providers/'));
+
+        if (this._registry.adminUserEnabled) {
+            try {
+                let creds = await client.registries.listCredentials(resourceGroup, this._registry.name);
+                this._serverPassword = creds.passwords[0].value;
+                this._serverUserName = creds.username;
+                return;
+            } catch (error) {
+                vscode.window.showErrorMessage(error.message);
+                throw new Error(('Admin not enabled'));
+            }
+        }
+        await vscode.window.showErrorMessage('Azure App service currently only supports running images from Azure Container Registries with admin enabled');
+        throw new Error(('Admin not enabled'));
+        // Un-Comment when Admin enabled is not required
+        // let convertedCredentials: { password: string, username: string } = await this.getSessionCredentials();
+        // this._serverUserName = convertedCredentials.password;
+        // this._serverPassword = convertedCredentials.username;
```

Comment:
Is the support of service principal ready?

Reply:
It can be done on our side and it will probably be added on a per feature thing as we add functionality such as pull and push but app services itself does not seem to support them at the moment.